mmv1/third_party/terraform/services/sql/resource_sql_database_instance.go.tmpl

```diff
@@ -1434,6 +1446,44 @@ func stringContainsSlice(arr []string, str string) bool {
 	return false
 }
 
+// enhancedBackupManagerCustomizeDiff suppresses diff changes to settings.backup_configuration
+// when the SQL instance is managed by Google Cloud Backup and Disaster Recovery (DR) Service.
+func enhancedBackupManagerCustomizeDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{}) error {
+	return EnhancedBackupManagerCustomizeDiffFunc(d)
+}
+
+func EnhancedBackupManagerCustomizeDiffFunc(d tpgresource.TerraformResourceDiff) error {
+    backupTierStr, ok := d.Get("settings.0.backup_configuration.0.backup_tier").(string)
+    if !ok || backupTierStr == "" {
+        return nil
+    }
+    if !strings.EqualFold(backupTierStr, "ENHANCED") {
+        return nil
+    }
+	// list of fields that must be ignored when ENHANCED backup manager is enabled
+	fieldsToIgnore := []string{
+		"settings.0.backup_configuration.0.enabled",
+		"settings.0.backup_configuration.0.start_time",
+		"settings.0.backup_configuration.0.binary_log_enabled",
+		"settings.0.backup_configuration.0.point_in_time_recovery_enabled",
+		"settings.0.backup_configuration.0.transaction_log_retention_days",
+		"settings.0.backup_configuration.0.backup_retention_settings.0.retained_backups",
+	}
+	for _, field := range fieldsToIgnore {
+		if d.HasChange(field) {
+			log.Printf(
+				"[INFO] Instance %s is managed by Google Cloud Backup and Disaster Recovery (BackupDR). "+
+					"Terraform will not manage the '%s' field. "+
+					"Any changes to this field in your Terraform configuration will be ignored.", d.Get("name"), field,
+			)
+			if err := d.Clear(field); err != nil {
```

Comment:
As I suspected, diff.Clear only impacts what's shown in the plan, not what's sent to the API. In addition to suppressing diffs (with Clear or DiffSuppressFuncs) you'll need to change the Update function so that it doesn't send these fields if they shouldn't be sent.

Reply:
As discussed offline, diff.Clear actually _does_ impact what's sent to the API. (But you may want to modify the update method anyway.)